knative-operator/pkg/apis/operator/v1alpha1/knativekafka_types.go

```diff
@@ -69,3 +69,19 @@ type Channel struct {
 func init() {
 	SchemeBuilder.Register(&KnativeKafka{}, &KnativeKafkaList{})
 }
+
+// Used in s-o e2e tests. can't instantiate there because vendor dirs
+// of s-o and s-o/knative-operator are different
```

Comment:
Something like this should fix the dependency chain: https://github.com/openshift-knative/serverless-operator/commit/13ab6e9b3f9cab8b1f81ab52830ca6a4d5306c38. We basically need to tell go.mod to not use the internet but just look right next door for the codez.

Reply:
Getting this:
```
Cannot use 'metav1.ObjectMeta{ Name: name, Namespace: namespace, }' (type "github.com/openshift-knative/serverless-operator/test/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta) as type "github.com/openshift-knative/serverless-operator/knative-operator/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta 
```

It is a similar problem, but slightly different. Go doesn't see the `metav1` in `vendor` and the `meta1` in `knative-operator/vendor` as the same thing.

I didn't try hard solving this frankly.

Reply:
You get this on the commit I shared? Interesting, it seemed to compile fine for me... looking again.

Reply:
Intellij still complains, but I think it is fine.
Created https://github.com/openshift-knative/serverless-operator/pull/577 to not interrupt the CI in this PR.